platform/linuxbsd/joypad_linux.cpp

```diff
@@ -340,6 +344,125 @@ void JoypadLinux::setup_joypad_properties(Joypad &p_joypad) {
 	}
 }
 
+void JoypadLinux::_auto_remap(Joypad &p_joypad, const StringName &p_guid, const String &p_name, bool p_hat0x_exist, bool p_hat0y_exist) {
+	if (p_joypad.key_map[BTN_GAMEPAD] == -1) {
+		return;
+	}
+
+	// Generate key mapping for JoyButton.
+
+	int joy_button_mappings[int(JoyButton::SDL_MAX)];
+	for (int i = 0; i < int(JoyButton::SDL_MAX); i++) {
+		joy_button_mappings[i] = -1;
+	}
+
+#define BUTTON_MAP_KEY(button, keycode) (joy_button_mappings[int(button)] = p_joypad.key_map[keycode])
+#define KEY_EXIST(keycode) (p_joypad.key_map[keycode] != -1)
+#define UNUSED_KEY_HIDE(keycode) (p_joypad.key_map[keycode] = -p_joypad.key_map[keycode] - 2) // Used for two events occur at one key press.
+
+	BUTTON_MAP_KEY(JoyButton::A, BTN_A);
+	BUTTON_MAP_KEY(JoyButton::B, BTN_B);
+	BUTTON_MAP_KEY(JoyButton::X, BTN_X);
+	BUTTON_MAP_KEY(JoyButton::Y, BTN_Y);
```

Comment:
Given that the spec seems to focus on the direction-based constants (ie `BTN_NORTH`, `BTN_SOUTH`, etc), perhaps this should use those instead?

(I also think that this mapping may be incorrect based on the directions, but I've already talked about that in the main comments :-))